Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Processes/ApiGatewayEmulatorProcess.cs

```diff
@@ -0,0 +1,78 @@
+using Amazon.Lambda.TestTool.Commands.Settings;
+using Amazon.Lambda.TestTool.Extensions;
+using Amazon.Lambda.TestTool.Models;
+using Amazon.Lambda.TestTool.Services;
+
+namespace Amazon.Lambda.TestTool.Processes;
+
+/// <summary>
+/// A process that runs the API Gatewat emulator.
+/// </summary>
+public class ApiGatewayEmulatorProcess
+{
+    /// <summary>
+    /// The service provider that will contain all the registered services.
+    /// </summary>
+    public required IServiceProvider Services { get; init; }
+
+    /// <summary>
+    /// The API Gatewat emulator task that was started.
+    /// </summary>
+    public required Task RunningTask { get; init; }
+
+    /// <summary>
+    /// The endpoint of the API Gatewat emulator.
+    /// </summary>
+    public required string ServiceUrl { get; init; }
+
+    /// <summary>
+    /// Creates the Web API and runs it in the background.
+    /// </summary>
+    public static ApiGatewayEmulatorProcess Startup(RunCommandSettings settings, CancellationToken cancellationToken = default)
+    {
+        var builder = WebApplication.CreateBuilder();
+
+        builder.Services.AddApiGatewayEmulatorServices();
+        
+        var serviceUrl = $"http://{settings.Host}:{settings.ApiGatewayEmulatorPort}";
+        builder.WebHost.UseUrls(serviceUrl);
+        builder.WebHost.SuppressStatusMessages(true);
+
+        builder.Services.AddHealthChecks();
+
+        var app = builder.Build();
+
+        app.UseHttpsRedirection();
+
+        app.MapHealthChecks("/health");
```

Comment:
If we want/need this we have to use a less common resource path to not interfere with any `/health` resource paths being emulated by the Lambda functions. Something like `/__lambda_test_tool_apigateway_health__`. But given we are web app for running just locally do really need this?

Reply:
The health check is needed purely for testing. I am using it to make sure that the process went up before I can cancel it. I renamed it to the name you suggested.